SourceX/dx.cpp

```diff
@@ -249,6 +250,77 @@ void LimitFrameRate()
 	frameDeadline = tc + v + refreshDelay;
 }
 
+#ifdef USE_SDL1
+
+void NewHWCursor(SDL_Surface *surf)
+{
+}
+
+#else
+
+static SDL_Surface *cursor_surface = NULL, *cursor_realsurface = NULL;
+static SDL_Cursor *cursor = NULL;
+static int cursor_width, cursor_height;
+static int cursor_surface_palette_version = -1;
+static int cursow_window_res_version = -1;
+extern BOOL oar;
+
+void NewHWCursor(SDL_Surface *surf)
+{
+	int scaled_w = surf->w, scaled_h = surf->h;
+
+	if (renderer) {
+		// upscale
+		int ww, wh;
+
+		SDL_GetWindowSize(ghMainWnd, &ww, &wh);
+
+		float hFactor = float(wh) / SCREEN_HEIGHT;
+		float wFactor = float(ww) / SCREEN_WIDTH;
+
+		if (wFactor > hFactor) {
+			scaled_w = int(scaled_w * hFactor);
+			scaled_h = int(scaled_h * hFactor);
+		} else {
+			scaled_w = int(scaled_w * wFactor);
+			scaled_h = int(scaled_h * wFactor);
+		}

```

Comment:
Can we simplify this taking the `Min(wFactor, hFactor)`? It seems to be the only difference in the formulas above.

Reply:
I don't think that would simplify anything.